src/providers/WorkspaceSymbolProvider.ts

```diff
@@ -1,85 +1,91 @@
+    " UNION ALL " +
+    "SELECT Name, Parent->ID AS Parent, 'trigger' AS Type FROM %Dictionary.TriggerDefinition" +
+    " UNION ALL " +
+    "SELECT Name, Parent->ID AS Parent, 'storage' AS Type FROM %Dictionary.StorageDefinition" +
+    " UNION ALL " +
+    "SELECT Name, Parent->ID AS Parent, 'projection' AS Type FROM %Dictionary.ProjectionDefinition" +
+    ") WHERE %SQLUPPER Name %MATCHES ?";
+
+  public provideWorkspaceSymbols(query: string): vscode.ProviderResult<vscode.SymbolInformation[]> {
+    if (query.length === 0) {
       return null;
     }
-    return Promise.all([this.byStudioDocuments(query), this.byMethods(query)]).then(([documents, methods]) => [
-      ...documents,
-      ...methods,
-    ]);
-  }
-
-  private getApi(): AtelierAPI {
-    const currentFileUri = vscode.window.activeTextEditor?.document.uri;
-    const firstFolder = vscode.workspace.workspaceFolders?.length ? vscode.workspace.workspaceFolders[0] : undefined;
-    return new AtelierAPI(currentFileUri || firstFolder?.uri || "");
-  }
-
-  private async byStudioDocuments(query: string): Promise<vscode.SymbolInformation[]> {
-    const searchAllDocTypes = config("searchAllDocTypes");
-    if (searchAllDocTypes) {
-      // Note: This query could be expensive if there are too many files available across the namespaces
-      // configured in the current vs code workspace. However, delimiting by specific file types
-      // means custom Studio documents cannot be found. So this is a trade off
-      query = `*${query}*`;
-    } else {
-      // Default is to only search classes, routines and include files
-      query = `*${query}*.cls,*${query}*.mac,*${query}*.int,*${query}*.inc`;
+    let pattern = "";
+    for (let i = 0; i < query.length; i++) {
+      const char = query.charAt(i);
+      pattern += char === "*" || char === "?" ? `*\\${char}` : `*${char}`;
     }
-    const sql = `SELECT TOP 10 Name FROM %Library.RoutineMgr_StudioOpenDialog(?,?,?,?,?,?,?)`;
-    const api = this.getApi();
-    const direction = "1";
-    const orderBy = "1";
-    const systemFiles = "1";
-    const flat = "1";
-    const notStudio = "0";
-    const generated = "0";
+    const workspace = currentWorkspaceFolder();
+    const api = new AtelierAPI(workspace);
+    return api.actionQuery(this.sql, [pattern.toUpperCase() + "*"]).then((data) => {
+      const result = [];
+      const uris: Map<string, vscode.Uri> = new Map();
+      for (const element of data.result.content) {
+        let kind: vscode.SymbolKind = vscode.SymbolKind.Property;
+        if (element.Type === "method" || element.Type === "query") {
+          kind = vscode.SymbolKind.Method;
+        } else if (element.Type === "parameter") {
+          kind = vscode.SymbolKind.Constant;
+        } else if (element.Type === "index") {
+          kind = vscode.SymbolKind.Key;
+        } else if (element.Type === "xdata" || element.Type === "storage") {
+          kind = vscode.SymbolKind.Struct;
+        }
```

Comment:
```suggestion
    const kind: vscode.SymbolKind = (() => {
      switch (element.type) {
        case "query":
        case "method":
          return vscode.SymbolKind.Method;
        case "parameter":
          return vscode.SymbolKind.Constant;
        case "index":
          return vscode.SymbolKind.Key;
        case "xdata":
        case "storage":
          return vscode.SymbolKind.Struct;
        case "property":
        default:
          return vscode.SymbolKind.Property;
      }
    })();
```
I think this will make it just more readable